cpp/arcticdb/storage/s3/detail-inl.hpp

```diff
@@ -240,8 +241,19 @@ namespace s3 {
 
             util::check(to_delete.empty(), "Have {} segment that have not been removed",
                         to_delete.size());
-            if (!failed_deletes.empty())
-                throw KeyNotFoundException(Composite<VariantKey>(std::move(failed_deletes)));
+            if (!failed_deletes.empty()){
+                auto failed_deletes_message = std::ostringstream();
+                for (auto i=0u; i<failed_deletes.size(); ++i){
+                    auto& failed = failed_deletes[i];
```

Comment:
If we keep the pair then consider expanding it with a structured binding here rather than `.first` and `.second`